lib/jsdom/living/postmessage.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+const isValidTargetOrigin = require("../utils").isValidTargetOrigin;
+
+module.exports = function (message, targetOrigin) {
+  let window = this;
+  let event;
+
+  if (!targetOrigin) {
+    throw new TypeError("'postMessage' requires a second argument 'targetOrigin'");
+  }
+
+  if (!isValidTargetOrigin(targetOrigin)) {
+    throw new SyntaxError("Failed to execute 'postMessage' on 'Window': " +
```

Comment:
Unfortunately not quite... I think it's `throw new DOMException(DOMException.SYNTAX_ERR, "Failed...")`.

You'll also need to require() the dom exception module, but thankfully that's nice and isolated these days.